sdk/storage/azure-storage-queues/inc/azure/storage/queues/queue_client.hpp

```diff
@@ -0,0 +1,126 @@
+    explicit QueueClient(
+        const std::string& queueUrl,
+        std::shared_ptr<Core::Credentials::TokenCredential> credential,
+        const QueueClientOptions& options = QueueClientOptions());
+
+    explicit QueueClient(
+        const std::string& queueUrl,
+        const QueueClientOptions& options = QueueClientOptions());
+
+    std::string GetUrl() const { return m_queueUrl.GetAbsoluteUrl(); }
+
+    Azure::Response<Models::CreateQueueResult> Create(
+        const CreateQueueOptions& options = CreateQueueOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::CreateQueueResult> CreateIfNotExists(
+        const CreateQueueOptions& options = CreateQueueOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::DeleteQueueResult> Delete(
+        const DeleteQueueOptions& options = DeleteQueueOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::DeleteQueueResult> DeleteIfExists(
+        const DeleteQueueOptions& options = DeleteQueueOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::QueueProperties> GetProperties(
+        const GetQueuePropertiesOptions& options = GetQueuePropertiesOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::SetQueueMetadataResult> SetMetadata(
+        Metadata metadata,
+        const SetQueueMetadataOptions& options = SetQueueMetadataOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::QueueAccessPolicy> GetAccessPolicy(
+        const GetQueueAccessPolicyOptions& options = GetQueueAccessPolicyOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::SetQueueAccessPolicyResult> SetAccessPolicy(
+        std::vector<Models::SignedIdentifier> signedIdentifiers,
+        const SetQueueAccessPolicyOptions& options = SetQueueAccessPolicyOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::SendMessageResult> SendMessage(
+        std::string messageText,
+        const SendMessageOptions& options = SendMessageOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::QueueMessage> ReceiveMessage(
+        const ReceiveMessageOptions& options = ReceiveMessageOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<std::vector<Models::QueueMessage>> ReceiveMessages(
+        const ReceiveMessagesOptions& options = ReceiveMessagesOptions(),
+        const Azure::Core::Context& context = Azure::Core::Context()) const;
+
+    Azure::Response<Models::PeekedQueueMessage> PeekMessage(
```

Comment:
It is not consistent to have `PeekedQueueMessage` from `PeekMessage` + `QueueMessage` from `ReceiveMessage`.
If we are honoring the rule at all, it should also be `QueueMessage` from `PeekMessage` or `PeekMessageResult`.
Or `ReceivedQueueMessage` from `ReceiveMessage`, which I personally think is too much.